### PR TITLE
QA-159: Safe Shutdown Takes 5 Seconds, which clips audio

### PR DESCRIPTION
**Pull request by cwinget**

### CODE REVIEW

**Review by toadzky — COMMENTED** [review]
i think we should have both graceful shutdown and immediate shutdown methods rather than changing the existing behavior.

**Comment by cwinget** [comment]
The entire reason for changing from graceful to immediate is to avoid the problem of slow shutdown cutting off audio, which in tests appears to work. Wouldn't putting the graceful back in just slow things down again and cause tests to fail? What's the downside of using the immediate shutdown only?

**Comment by toadzky** [comment]
allowing the consumer to choose makes sense. this is why netty, jdk threadpools, etc all have both: it's up to the user to decide how to use it. forcing the shutdown to be immediate in all cases is less flexible and could cause issues with tail clipping.

File: src/main/java/com/biasedbit/efflux/session/AbstractRtpSession.java [hunk 1]
```diff
@@ -802,7 +807,7 @@ protected SourceDescriptionPacket buildSdesPacket(long currentSsrc) {
         return sdesPacket;
     }
 
-    protected synchronized void terminate(Throwable cause) {
+    protected synchronized void terminate(Throwable cause, Boolean graceful) {
```
Comment (toadzky) [inline src/main/java/com/biasedbit/efflux/session/AbstractRtpSession.java]: use lowercase `boolean` so it can't be null

**Review by toadzky — APPROVED** [review]
fix the 1 nit then :shipit:

**Pull request merged**